safe_cli/operators/hw_accounts/hw_account_manager.py

```diff
@@ -0,0 +1,140 @@
+from enum import Enum
+from typing import Dict, List, Optional, Set, Tuple
+
+from eth_typing import ChecksumAddress
+from prompt_toolkit import HTML, print_formatted_text
+
+from gnosis.eth.eip712 import eip712_encode
+from gnosis.safe import SafeTx
+
+from safe_cli.operators.hw_accounts.hw_account import HwAccount
+
+
+class HwWalletType(Enum):
+    TREZOR = "Trezor"
+    LEDGER = "Ledger"
+
+
+class HwAccountManager:
+    def __new__(cls):
+        if not hasattr(cls, "instance"):
+            cls.instance = super(HwAccountManager, cls).__new__(cls)
+        return cls.instance
+
+    def __init__(self):
+        self.accounts: Set[HwAccount] = set()
+        self.supported_hw_wallets: Dict[str, HwAccount] = {}
+        try:
+            from safe_cli.operators.hw_accounts.ledger_manager import LedgerManager
+
+            self.supported_hw_wallets[HwWalletType.LEDGER] = LedgerManager
+        except (ModuleNotFoundError, IOError):
+            pass
+
+        try:
+            from safe_cli.operators.hw_accounts.trezor_manager import TrezorManager
+
+            self.supported_hw_wallets[HwWalletType.TREZOR] = TrezorManager
+        except (ModuleNotFoundError, IOError):
+            pass
+
+    def is_supported_hw_wallet(self, hw_wallet_type: HwWalletType):
+        return hw_wallet_type in self.supported_hw_wallets
+
+    def get_hw_wallet(self, hw_wallet_type: HwWalletType):
+        if hw_wallet_type in self.supported_hw_wallets:
+            return self.supported_hw_wallets[hw_wallet_type]
+        # TODO add unsupported exception
```

Comment:
Are you adding this exception in this PR or this is for a future refactor?

Reply:
It is not necessary now, as the user haven't option to choose unsupported wallet type.

Reply:
If it's not needed then remove the TODO, otherwise it looks like it's something needed not implemented yet 😉